Lombiq.HelpfulExtensions.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs

```diff
@@ -105,17 +105,34 @@ public static async Task TestFeatureCodeGenerationsAsync(this UITestContext cont
         await context.EnableFeatureDirectlyAsync(FeatureIds.CodeGeneration);
 
         await context.GoToContentTypeEditorAsync("Page");
-        await context.ClickReliablyOnAsync(By.ClassName("toggle-showing-generated-migration-code"));
 
-        context.Get(By.Id("generated-migration-code").OfAnyVisibility()).GetValue().ShouldBe(GeneratedMigrationCodes.Page);
-
-        // Making sure that the collapsible area is open.
-        context.Get(By.CssSelector("#generated-migration-code-container.collapse.show"));
-
-        // Checking the first line of the CodeMirror editor.
-        context.Get(By.CssSelector(".CodeMirror-line .cm-variable")).Text.ShouldBe("_contentDefinitionManager");
-        context.Get(By.CssSelector(".CodeMirror-line .cm-property")).Text.ShouldBe("AlterTypeDefinition");
-        context.Get(By.CssSelector(".CodeMirror-line .cm-string")).Text.ShouldBe("\"Page\"");
+        // These CodeMirror checks can throw StaleElementReferenceException even if the code section is there, opened,
+        // and visible in the current viewport (though this shouldn't matter). So, it should just work, but it doesn't,
+        // so we need to retry for these random issues.
+        await context.RetryIfStaleOrFailAsync(async () =>
+        {
+            try
+            {
+                await context.ClickReliablyOnAsync(By.ClassName("toggle-showing-generated-migration-code"));
+
+                context.Get(By.Id("generated-migration-code").OfAnyVisibility()).GetValue().ShouldBe(GeneratedMigrationCodes.Page);
+
+                // Making sure that the collapsible area is open.
+                context.Get(By.CssSelector("#generated-migration-code-container.collapse.show"));
+
+                // Checking the first line of the CodeMirror editor.
+                context.Get(By.CssSelector(".CodeMirror-line .cm-variable")).Text.ShouldBe("_contentDefinitionManager");
+                context.Get(By.CssSelector(".CodeMirror-line .cm-property")).Text.ShouldBe("AlterTypeDefinition");
+                context.Get(By.CssSelector(".CodeMirror-line .cm-string")).Text.ShouldBe("\"Page\"");
+
+                return true;
+            }
+            catch (StaleElementReferenceException)
+            {
+                context.Refresh();
+                throw;
+            }
+        });
```

Comment:
BTW This approach does not seem good, since the click on the button toggles the `generated-migration-code-container` visibility. So If the first try any element gets stale, then in the next cycle, the button gets clicked again which possibly removes the `show` css class and the `generated-migration-code-container` get hidden, after this the `context. Get` will throw a not found exception and the test fail.

Reply:
A possible solution is to move out `await context.ClickReliablyOnAsync(By.ClassName("toggle-showing-generated-migration-code"));` of the cycle, then it will be called only once while the other checks retried in case of error.

Reply:
The click needs to be done every time to open the code area, see my comment above https://github.com/Lombiq/Helpful-Extensions/pull/109#discussion_r1067488529. So it needs to be repeated after a refresh. The refresh in turn is needed so if anything is stale (what it really shouldn't), it gets resolved.

This works in a loop, I checked.

Reply:
Yes, sorry, I mixed it with an other `context.Refresh`